File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/ui/CinacoinChainSelector.kt
package com.cinacoin.appkit.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.cinacoin.appkit.config.ChainRegistry
import com.cinacoin.appkit.config.CinacoinTheme
import com.cinacoin.appkit.config.model.ChainConfig

/**
 * Chain selector dropdown
 */
@Composable
fun CinacoinChainSelector(
    selectedChainId: Int,
    onChainSelected: (Int) -> Unit,
    chains: List<ChainConfig> = ChainRegistry.allEVMChains,
    theme: CinacoinTheme = if (androidx.compose.foundation.isSystemInDarkTheme()) 
        CinacoinTheme.Dark else CinacoinTheme.Light
) {
    var expanded by remember { mutableStateOf(false) }
    val selectedChain = chains.firstOrNull { it.chainId == selectedChainId }
    
    Box {
        Row(
            modifier = Modifier
                .clip(RoundedCornerShape(20.dp))
                .background(theme.surface)
                .border(1.dp, theme.border, RoundedCornerShape(20.dp))
                .clickable { expanded = true }
                .padding(horizontal = 12.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            if (selectedChain != null) {
                AsyncImage(
                    model = selectedChain.iconUrl,
                    contentDescription = selectedChain.name,
                    modifier = Modifier
                        .size(20.dp)
                        .clip(CircleShape)
                )
                
                Spacer(modifier = Modifier.width(8.dp))
                
                Text(
                    text = selectedChain.shortName.uppercase(),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    color = theme.text
                )
            }
            
            Icon(
                imageVector = Icons.Default.KeyboardArrowDown,
                contentDescription = null,
                tint = theme.textSecondary,
                modifier = Modifier.size(16.dp)
            )
        }
        
        DropdownMenu(
            expanded = expanded,
            onDismissRequest = { expanded = false }
        ) {
            chains.forEach { chain ->
                DropdownMenuItem(
                    text = {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            AsyncImage(
                                model = chain.iconUrl,
                                contentDescription = chain.name,
                                modifier = Modifier
                                    .size(20.dp)
                                    .clip(CircleShape)
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(chain.name)
                        }
                    },
                    onClick = {
                        onChainSelected(chain.chainId)
                        expanded = false
                    },
                    trailingIcon = {
                        if (chain.chainId == selectedChainId) {
                            Text("✓", color = theme.primary)
                        }
                    }
                )
            }
        }
    }
}

File: packages/flutter-dart/lib/appkit_ui/cinacoin_chain_selector.dart
import 'package:flutter/material.dart';
import '../appkit_config/chain_registry.dart';
import '../appkit_config/cinacoin_theme.dart';
import '../appkit_config/models/chain_config.dart';

/// Chain selector dropdown
class CinacoinChainSelector extends StatelessWidget {
  final int selectedChainId;
  final void Function(int chainId) onChainSelected;
  final List<ChainConfig> chains;
  final CinacoinTheme? theme;

  const CinacoinChainSelector({
    super.key,
    required this.selectedChainId,
    required this.onChainSelected,
    this.chains = const [],
    this.theme,
  });

  @override
  Widget build(BuildContext context) {
    final themeData = theme ?? _getThemeFromBrightness(context);
    final chainList = chains.isEmpty ? ChainRegistry.allEVMChains : chains;
    final selectedChain = chainList.firstWhere(
      (c) => c.chainId == selectedChainId,
      orElse: () => ChainRegistry.ethereum,
    );

    return PopupMenuButton<int>(
      onSelected: onChainSelected,
      offset: const Offset(0, 40),
      color: themeData.surface,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(12),
        side: BorderSide(color: themeData.border),
      ),
      child: Container(
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
        decoration: BoxDecoration(
          color: themeData.surface,
          borderRadius: BorderRadius.circular(20),
          border: Border.all(color: themeData.border),
        ),
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            if (selectedChain.iconUrl != null)
              CircleAvatar(
                radius: 10,
                backgroundImage: NetworkImage(selectedChain.iconUrl!),
                backgroundColor: themeData.surface,
              ),
            const SizedBox(width: 8),
            Text(
              selectedChain.shortName.toUpperCase(),
              style: TextStyle(
                fontSize: 14,
                fontWeight: FontWeight.w500,
                color: themeData.text,
              ),
            ),
            const SizedBox(width: 4),
            Icon(
              Icons.keyboard_arrow_down,
              size: 16,
              color: themeData.textSecondary,
            ),
          ],
        ),
      ),
      itemBuilder: (context) {
        return chainList.map((chain) {
          return PopupMenuItem<int>(
            value: chain.chainId,
            child: Row(
              children: [
                if (chain.iconUrl != null)
                  CircleAvatar(
                    radius: 10,
                    backgroundImage: NetworkImage(chain.iconUrl!),
                  ),
                const SizedBox(width: 8),
                Text(
                  chain.name,
                  style: TextStyle(color: themeData.text),
                ),
                const Spacer(),
                if (chain.chainId == selectedChainId)
                  Icon(Icons.check, color: themeData.primary, size: 16),
              ],
            ),
          );
        }).toList();
      },
    );
  }

  CinacoinTheme _getThemeFromBrightness(BuildContext context) {
    final brightness = Theme.of(context).brightness;
    return brightness == Brightness.dark 
        ? CinacoinTheme.dark 
        : CinacoinTheme.light;
  }
}
